fw/ss_skb.c

```diff
@@ -1708,12 +1709,21 @@ ss_skb_cut_extra_data(struct sk_buff *skb_head, struct sk_buff *skb,
 		       " len=%d, tail=%d\n", __func__, frag_num, size, offset,
 		       stop_offset, len, tail);
 
-		if (frag_num >= 0)
+		if (frag_num >= 0) {
 			ret = __split_pgfrag_del(skb_head, skb, frag_num,
 						 offset, len, &it);
-		else
+			/*
+			 * split_pgfrag_del() can delete whole fragment and
+			 * then decrease @skb->nr_frags, therefore we need to
+			 * update @frag_num.
+			 */
+			frag_num -= fragn - skb_shinfo(skb)->nr_frags;
```

Comment:
It seems the name `frag_num` is confusing since it's actually fragment ID, the fragment where we need to start to search. Suppose we have skb with 5 fragments and frag_num = 2. If `__split_pgfrag_del()` deletes the fragment, then fragment 3 is moved to position 2 and 4 to 3. Thus, we continue to work with the same frament ID, which points to a new fragment.

Reply:
Yes, I think along similar lines. Also, `fragn` is one more name you need to remember reading this code. I would just name it `nr_frags` to assert the origin and the relevance.

Reply:
As for me rename `fragn` to `nr_frags` is good idea. And I suggest to rename `frag_num` to `frag_index`, but for consistency with other part of code, we should use name `i`.